python_modules/dagster/dagster_tests/core_tests/definitions_tests/test_definitions.py

```diff
@@ -25,6 +27,71 @@ def test_deps_equal():
     assert DependencyDefinition('foo', 'bar') != DependencyDefinition('foo', 'quuz')
 
 
+def test_solid_def():
+    @lambda_solid
+    def produce_string():
+        return 'foo'
+
+    @solid(
+        inputs=[InputDefinition('input_one', types.String)],
+        outputs=[OutputDefinition(types.Any)],
+        config_field=Field(Dict({'another_field': Field(types.Int)})),
+    )
+    def solid_one(_context, input_one):
+        raise Exception('should not execute')
+
+    pipeline_def = PipelineDefinition(
+        solids=[produce_string, solid_one],
+        dependencies={'solid_one': {'input_one': DependencyDefinition('produce_string')}},
+    )
+
+    assert len(pipeline_def.solids[0].output_handles()) == 1
+
+    assert isinstance(pipeline_def.solid_named('solid_one'), Solid)
+
+    solid_one_solid = pipeline_def.solid_named('solid_one')
+
+    assert solid_one_solid.has_input('input_one')
+
+    assert isinstance(solid_one_solid.input_def_named('input_one'), InputDefinition)
+
+    assert len(solid_one_solid.input_defs) == 1
+    assert len(solid_one_solid.output_defs) == 1
+
+    # These aren't python version independent because dict orders are different
+    # assert str(solid_one_solid.input_handle('input_one')) == (
```

Comment:
?

Reply:
I can actually probably fix this by changing the way we generate the reprs